core/trino-main/src/main/java/io/trino/operator/PagesIndex.java

```diff
@@ -515,16 +515,21 @@ public LookupSourceSupplier createLookupSourceSupplier(
             // This code path will trigger only for OUTER joins. To fix that we need to add support for
             //        OUTER joins into NestedLoopsJoin and remove "type == INNER" condition in LocalExecutionPlanner.visitJoin()
 
-            LookupSourceSupplierFactory lookupSourceFactory = joinCompiler.compileLookupSourceFactory(types, joinChannels, sortChannel, outputChannels);
-            return lookupSourceFactory.createLookupSourceSupplier(
-                    session,
-                    valueAddresses,
-                    channels,
-                    hashChannel,
-                    filterFunctionFactory,
-                    sortChannel,
-                    searchFunctionFactories,
-                    hashArraySizeSupplier);
+            try {
+                LookupSourceSupplierFactory lookupSourceFactory = joinCompiler.compileLookupSourceFactory(types, joinChannels, sortChannel, outputChannels);
+                return lookupSourceFactory.createLookupSourceSupplier(
+                        session,
+                        valueAddresses,
+                        channels,
+                        hashChannel,
+                        filterFunctionFactory,
+                        sortChannel,
+                        searchFunctionFactories,
+                        hashArraySizeSupplier);
+            }
+            catch (Exception e) {
+                log.error(e, "Lookup source compile failed for types=%s error=%s", types, e);
+            }
         }
 
         PagesHashStrategy hashStrategy = new SimplePagesHashStrategy(
```

Comment:
We should remove fallback instead (see: https://github.com/trinodb/trino/commit/1232d3945dd8470ba7b2cc5f7e905bb750c56d09). Why do you want to restore it? Is there a compilation failure? In case compilation failed user will get a perf hit without even knowing that

Reply:
By this  https://github.com/trinodb/trino/blob/6a192cf46ef93723b390bbc161a2a3553d7e2a3a/core/trino-main/src/main/java/io/trino/operator/PagesIndex.java#L451 I thought slow performance might be allowed than a failure. 

A session parameter to allow fallback would be nice so that user can decide.

Reply:
That is a different code path for `PagesHashStrategy`. We should remove it too (need to check before if we compile all cases).

> A session parameter to allow fallback would be nice so that user can decide.

I'm not sure. It's really technical, so unless we see actual issues around that, I would not add a parameter

Reply:
ok, if I understand correctly, Trino would not allow queries causing compilation failure so that users could notice it and rewrite their failed queries.